src/main/java/com/ifmo/epampractice/daoImpl/InvoiceDaoImpl.java

```diff
@@ -0,0 +1,124 @@
+package com.ifmo.epampractice.daoImpl;
+
+import com.ifmo.epampractice.dao.DBConnectorInterface;
+import com.ifmo.epampractice.dao.DBConnectorPostgres;
+import com.ifmo.epampractice.dao.InvoiceDao;
+import com.ifmo.epampractice.entity.InvoiceEntity;
+import com.ifmo.epampractice.enums.InvoiceStatus;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.math.BigDecimal;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class InvoiceDaoImpl implements InvoiceDao {
+    private static final String GET_ALL_QUERY = "SELECT * FROM INVOICE";
+    private static final String GET_QUERY = "SELECT * FROM INVOICE WHERE ID = ?";
+    private static final String SAVE_QUERY = "INSERT INTO INVOICE (ORDER_ID, ISSUE_DATE, " +
+            "PAYMENT_DATE, TOTAL_PRICE, STATUS) VALUES ( ?, ?, ?, ?, ?::e_status_invoice)";
+    private static final String UPDATE_QUERY = "UPDATE INVOICE SET (ORDER_ID, ISSUE_DATE, " +
+            "PAYMENT_DATE, TOTAL_PRICE, STATUS) = (?, ?, ?, ?, ?::e_status_invoice) WHERE ID = ?";
+    private static final String DELETE_QUERY = "DELETE FROM INVOICE WHERE ID = ?";
+
+    private static final Logger log = LogManager.getLogger(InvoiceDaoImpl.class);
+    private DBConnectorInterface dbConnector = DBConnectorPostgres.getInstance();
+
+    @Override
+    public List<InvoiceEntity> getAll() {
+        List<InvoiceEntity> invoices = new ArrayList<>();
+        try (Connection connection = dbConnector.getConnection();
+             Statement statement = connection.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE,
+                     ResultSet.CONCUR_READ_ONLY);
+             ResultSet resultSet = statement.executeQuery(GET_ALL_QUERY)) {
+            while (resultSet.next()) {
+                InvoiceEntity currentInvoice = parseRow(resultSet);
+                invoices.add(currentInvoice);
+            }
+            return invoices;
+        } catch (SQLException e) {
+            log.error(e);
+            throw new RuntimeException(e);
```

Comment:
I think logger will be enough for us, because it is not a good idea to throw a generic exception